app/hooks/use-number-input.ts

```diff
@@ -0,0 +1,107 @@
+import { useState } from 'react';
+import type { NumpadButton } from '~/components/numpad';
+import { type Currency, Money } from '~/lib/money';
+import { getUnit } from '~/utils';
+import { useExchangeRate } from './use-exchange-rate';
+
+export type InputState = {
+  value: string;
+  convertedValue: string;
+  currency: Currency;
```

Comment:
I would maybe call this valueCurrency and convertedValueCurrency or maybe even organize it like this:
```
type InputState = {
  active: { value: string; currency: string }
  other: { value: string; currency: string }
}
```